pkg/apis/autoscaling/v1alpha1/pa_lifecycle.go

```diff
@@ -80,7 +81,7 @@ func (pa *PodAutoscaler) Target() (target int32, ok bool) {
 // IsReady looks at the conditions and if the Status has a condition
 // PodAutoscalerConditionReady returns true if ConditionStatus is True
 func (rs *PodAutoscalerStatus) IsReady() bool {
-	return podCondSet.Manage(rs).IsHappy()
+	return podCondSet.Manage((*duckv1beta1.Status)(rs)).IsHappy()
```

Comment:
Why do we need the cast here?
`rs` does implement ConditionsAccessor?

Reply:
As discussed.

Reply:
Yeah, tl;dr I added this because of Go type nonsense.  If we can find a better way, I'm open, but this was the closest approximation I could manage.